data-reporting-service/observation-service/src/main/java/gov/cdc/etldatapipeline/observation/util/ProcessObservationDataUtil.java

```diff
@@ -0,0 +1,158 @@
+package gov.cdc.etldatapipeline.observation.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cdc.etldatapipeline.observation.repository.model.Observation;
+import gov.cdc.etldatapipeline.observation.repository.model.ObservationTransformed;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class ProcessObservationDataUtil {
+    private static final Logger logger = LoggerFactory.getLogger(ProcessObservationDataUtil.class);
+    public ObservationTransformed transformObservationData(Observation observation) {
+        ObservationTransformed observationTransformed = new ObservationTransformed();
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        String obsDomainCdSt1 = observation.getObsDomainCdSt1();
+
+        transformPersonParticipations(observation.getPersonParticipations(), obsDomainCdSt1, observationTransformed, objectMapper);
+        transformOrganizationParticipations(observation.getOrganizationParticipations(), obsDomainCdSt1, observationTransformed, objectMapper);
+        transformMaterialParticipations(observation.getMaterialParticipations(), obsDomainCdSt1, observationTransformed, objectMapper);
+        transformFollowupObservations(observation.getFollowupObservations(), obsDomainCdSt1, observationTransformed, objectMapper);
+
+        return observationTransformed;
+    }
+
+    private void transformPersonParticipations(String personParticipations, String obsDomainCdSt1, ObservationTransformed observationTransformed, ObjectMapper objectMapper) {
+        try {
+            JsonNode personParticipationsJsonArray = personParticipations != null ? objectMapper.readTree(personParticipations) : null;
```

Comment:
Stored proc data is already stored in the class named `Investigation.java`. I'm trying to avoid the object creation as much as possible. But we can discuss this offline